### PR TITLE
feat: add hook for schemathesis tests

### PR DESCRIPTION
**Pull request by SotaTek-DuyLe**
# Description
- Add hook to modify schema so that schemathesis tries to do a POST on an endpoint that's only a GET and vice versa (PUT, PATCH, DELETE as well)

## Related Issue(s)

List the issue numbers related to this pull request.

> e.g., Closes #1, Resolves #1  Fixes #1

## Description of Changes

Provide a clear and concise description of what the pull request changes.

## Breaking Changes

Describe any breaking changes and the impact.

## Screenshots

If applicable, add screenshots to help explain your changes.

## Related Pull Requests

If applicable, list any related pull requests.

> e.g., #1, #1

## Please confirm the following checks

* [ ] My code follows the style guidelines of this project
* [ ] I have performed a self-review of my code
* [ ] I have commented my code, particularly in hard-to-understand areas
* [ ] I have made corresponding changes to the documentation
* [ ] My changes generate no new warnings
* [ ] I have added tests that prove my fix is effective or that my feature works
* [ ] New and existing unit tests pass locally with my changes
* [ ] Any dependent changes have been merged and published in downstream module

### CODE REVIEW

File: catalyst-gateway/tests/api_tests/api_tests/hooks.py
Comment (kukkok3) [inline catalyst-gateway/tests/api_tests/api_tests/hooks.py:1]: I would create a folder called schemathesis-tests in tests/ and put all the schemathesis stuff in there (the earthfile, the hooks etc) wdyt?

Comment (SotaTek-DuyLe) [reply catalyst-gateway/tests/api_tests/api_tests/hooks.py:1]: I think it's ok. So we will move schemathesis test from tests/ Earthfile to a new file right? Like in api_tests folder

**Review by kukkok3 — DISMISSED** [review]
LGTM

**Review by kukkok3 — APPROVED** [review]
LGTM

**Pull request merged**